algo/src/main/java/org/neo4j/graphalgo/metaPathComputationProcs/GetSchemaProc.java

```diff
@@ -32,26 +37,49 @@
     @Description("CALL algo.GetSchema() YIELD schema: \n" +
             "return schema as adjacency list \n")
 
-    public Stream<GetSchemaResult> GetSchema() {
+    public Stream<GetSchemaResult> GetSchema() throws FileNotFoundException {
 
         final GetSchemaResult.Builder builder = GetSchemaResult.builder();
-        final HeavyGraph graph;
 
-        graph = (HeavyGraph) new GraphLoader(api)
-                .asUndirected(true)
-                .withLabelAsProperty(true)
-                .load(HeavyGraphFactory.class);
+        ArrayList<HashSet<Pair>> schema = null;
+        HashMap<Integer, Integer> reversedLabelDictionary = null;
+        boolean notYetComputed = false;
+        try {
+            FileInputStream fileIn = new FileInputStream("metagraph.ser");
+            ObjectInputStream in = new ObjectInputStream(fileIn);
+            schema = (ArrayList<HashSet<Pair>>) in.readObject();
+            in.close();
+            fileIn.close();
 
-        final GetSchema algo = new GetSchema(graph);
+            fileIn = new FileInputStream("reversedLabelDictionary.ser");
+            in = new ObjectInputStream(fileIn);
+            reversedLabelDictionary = (HashMap<Integer, Integer>) in.readObject();
+        } catch (IOException i) {
```

Comment:
Is there no way an other IOException could appear?

Reply:
What do you mean? If we can't read from the sar we need to compute the schema no matter what IOException is the problem.

Reply:
Maybe it was just an one time not deterministic thread based IOException.